Added info for JSON and XML config assemblies

nCubed opened the pull request:
The updates to Microsoft.Extensions.Configuration separates JSON and XML assemblies. This update makes it clear where to get the assemblies.

nCubed committed:
Added info for JSON and XML config assemblies

The updates to Microsoft.Extensions.Configuration separates JSON and XML assemblies. This update makes it clear where to get the assemblies.

tillig commented on docs/configuration/xml.rst:
```diff
@@ -26,6 +26,8 @@ Quick Start
 
 1. Set up your configuration in JSON or XML files that can be read by ``Microsoft.Extensions.Configuration``.
+  * JSON Configuration uses ``Microsoft.Extensions.Configuration.Json``
+  * XML Configuration uses ``Microsoft.Extensions.Configuration.XML``
```
It should be `Microsoft.Extensions.Configuration.Xml` (the "XML" in the library is cased like "Xml" for the NuGet package).

tillig replied:
"Configuration" should be lower case - "JSON configuration uses..." and "XML configuration uses..."

tillig commented on docs/configuration/xml.rst:
```diff
@@ -82,6 +84,8 @@ Build up your configuration and register it with the Autofac ``ContainerBuilder`
     // Add the configuration to the ConfigurationBuilder.
     var config = new ConfigurationBuilder();
+    // config.AddJsonFile comes from Microsoft.Extensions.Configuration.Json
+    // config.AddXmlFile comes from Microsoft.Extensions.Configuration.XML
```
`Microsoft.Extensions.Configuration.Xml`

nCubed committed:
Corrected casing of XML nuget name.

Corrected reference of Microsoft.Extensions.Configuration.XML to Microsoft.Extensions.Configuration.Xml.

nCubed commented:
I've updated the casing. This should match the nuget package names now for both Json and Xml.

tillig commented on docs/configuration/xml.rst:
```diff
@@ -27,7 +27,7 @@ The basic steps to getting configuration set up with your application are:
 1. Set up your configuration in JSON or XML files that can be read by ``Microsoft.Extensions.Configuration``.
   * JSON Configuration uses ``Microsoft.Extensions.Configuration.Json``
-  * XML Configuration uses ``Microsoft.Extensions.Configuration.XML``
+  * XML Configuration uses ``Microsoft.Extensions.Configuration.Xml``
```
Still need "configuration" in the sentence ("XML configuration" and "JSON configuration") to be lower case (not "JSON Configuration" - "Configuration" isn't a proper noun).

nCubed replied:
Good catch!

nCubed committed:
Update casing of Configuration for correct grammar

nCubed commented:
If I have another pull request, I'll be more cognizant of grammar and structure. I believe we should be good to go now.

tillig commented:
Thanks very much for the contribution! Sorry to seem pedantic, and very much appreciate your patience.

nCubed commented:
TBH, I appreciate you being so pedantic; I'm the same way. I think it takes more patience on your side than mine to ensure we contribute to the project in a manner consistent with the project owner's vision and style.
